tests/tape/tapes/test_jacobian_tape.py

```diff
@@ -541,14 +541,14 @@ def test_multiple_output_values(self, tol):
         with JacobianTape() as tape:
             qml.ThermalState(n, wires=0)
             qml.Displacement(a, 0, wires=0)
-            qml.expval(qml.NumberOperator(0))
+            qml.expval(qml.NumberOperator(1))
             qml.var(qml.NumberOperator(0))
 
         tape.trainable_params = {0, 1}
         res = tape.jacobian(dev)
         assert res.shape == (2, 2)
 
-        expected = np.array([[1, 2 * a], [2 * a ** 2 + 2 * n + 1, 2 * a * (2 * n + 1)]])
+        expected = np.array([[0, 0], [2 * a ** 2 + 2 * n + 1, 2 * a * (2 * n + 1)]])
```

Comment:
It actually surprised me that this was working, since I got incorrect results when evaluating, e.g., `expval(qml.PauliZ(0)), expval(qml.PauliX(1))`, which returned two copies of the final expval. This is due to how `statistics()` are handled in `QubitDevice.execute()`, which performs a rotation based upon diagonalizing gates that only pick up the `PauliX` term.

However, the above works because it is using `default.gaussian`, which inherits from `Device` and implements `execute()` differently.